src/include/duckdb/execution/window_executor.hpp

```diff
@@ -169,148 +156,173 @@ class WindowExecutor {
 
 	// evaluate RANGE expressions, if needed
 	WindowInputColumn range;
+};
 
-	virtual void EvaluateInternal(WindowExecutorState &lstate, Vector &result, idx_t count, idx_t row_idx) const = 0;
+class WindowExecutorLocalState : public WindowExecutorState {
+public:
+	explicit WindowExecutorLocalState(const WindowExecutorGlobalState &gstate) {
+	}
 };
 
-class WindowAggregateExecutor : public WindowExecutor {
+class WindowExecutor {
 public:
-	bool IsConstantAggregate();
-	bool IsCustomAggregate();
-	bool IsDistinctAggregate();
+	WindowExecutor(BoundWindowExpression &wexpr, ClientContext &context);
+	virtual ~WindowExecutor() {
+	}
 
-	WindowAggregateExecutor(BoundWindowExpression &wexpr, ClientContext &context, const idx_t payload_count,
-	                        const ValidityMask &partition_mask, const ValidityMask &order_mask,
-	                        WindowAggregationMode mode);
+	virtual unique_ptr<WindowExecutorGlobalState>
+	GetGlobalState(const idx_t payload_count, const ValidityMask &partition_mask, const ValidityMask &order_mask) const;
```

Comment:
Yes multiple global states can be created (one per hash group) and right now we have one per thread. The long term goal of the project is to have as few in flight at once as they can be memory hogs.

Reply:
I get what you are saying about the names (it is a bit strange) but I took them from the physical operator naming.

Reply:
If multiples can exist simultaneously, the name makes more sense to me since one can 'get' as many as one needs. And fair, consistency with other places is more critical here.